Userland/Libraries/LibWeb/HTML/Canvas/CanvasPathClipper.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, MacDue <macdue@dueutil.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGfx/Painter.h>
+#include <LibGfx/Path.h>
+
+namespace Web::HTML {
+
+struct CanvasClip {
+    Gfx::Path path;
+    Gfx::Painter::WindingRule winding_rule;
+};
+
+class CanvasPathClipper {
+public:
+    static ErrorOr<CanvasPathClipper> create(Gfx::Painter&, CanvasClip const& canvas_clip);
+    ErrorOr<void> apply_clip(Gfx::Painter& painter);
+
+private:
+    CanvasPathClipper(RefPtr<Gfx::Bitmap const> saved_clip_region, Gfx::IntRect bounding_box, CanvasClip const& canvas_clip)
+        : m_saved_clip_region(saved_clip_region)
+        , m_bounding_box(bounding_box)
+        , m_canvas_clip(canvas_clip)
+    {
+    }
+
+    RefPtr<Gfx::Bitmap const> m_saved_clip_region;
+    Gfx::IntRect m_bounding_box;
+    CanvasClip const& m_canvas_clip;
+};
+
+struct ScopedCanvasPathClip {
```

Comment:
Please use `class` for data structures that aren't just a bunch of values bundled together :)